kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerRuntimeManagementIntegrationTest.java

```diff
@@ -0,0 +1,128 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.server.integrationtests.controller;
+
+import static org.assertj.core.api.Assertions.*;
+import org.assertj.core.api.SoftAssertions;
+
+import static org.junit.Assert.fail;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.controller.api.model.runtime.Container;
+import org.kie.server.controller.api.model.runtime.ContainerList;
+import org.kie.server.controller.api.model.runtime.ServerInstanceKey;
+import org.kie.server.controller.api.model.runtime.ServerInstanceKeyList;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.client.exception.KieServerControllerClientException;
+import org.kie.server.controller.impl.storage.InMemoryKieServerTemplateStorage;
+import org.kie.server.integrationtests.shared.KieServerAssert;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+import org.kie.server.integrationtests.shared.KieServerSynchronization;
+
+public abstract class KieControllerRuntimeManagementIntegrationTest<T extends KieServerControllerClientException> extends KieControllerManagementBaseTest {
+
+    private KieServerInfo kieServerInfo;
+
+    protected abstract void assertNotFoundException(T exception);
+
+    protected abstract void assertBadRequestException(T exception);
+
+    @BeforeClass
+    public static void initialize() throws Exception {
+        KieServerDeployer.createAndDeployKJar(RELEASE_ID);
+
+    }
+
+    @Before
+    public void getKieServerInfo() {
+        InMemoryKieServerTemplateStorage.getInstance().clear();
+        // Getting info from currently started kie server.
+        ServiceResponse<KieServerInfo> reply = client.getServerInfo();
+        KieServerAssert.assertSuccess(reply);
+        kieServerInfo = reply.getResult();
+    }
+
+    @Test
+    public void testGetServerInstances() {
+        // Create kie server template connection in controller.
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        ServerInstanceKeyList serverInstances = controllerClient.getServerInstances(serverTemplate.getId());
+        assertThat(serverInstances.getServerInstanceKeys()).hasSize(1);
+        ServerInstanceKey serverInstance = serverInstances.getServerInstanceKeys()[0];
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(serverInstance.getServerName()).isEqualTo(kieServerInfo.getName());
```

Comment:
@cristianonicolai What is expected name of the server instance? I think it should be same as name of the Kie Server or at least server instance should contain it. But in this test failed with message: expected:<"[KieServer@/kie-server-services]"> but was:<"[a0e625bf-251d-3c1f-ad11-0a17f1c14e21@localhost:37668]">